python/res/enkf/site_config.py

```diff
@@ -13,48 +13,96 @@
 
 
-    def __init__(self, user_config_file=None, config_content=None):
+    def __init__(self, user_config_file=None, config_content=None, config_dict=None):
 
+        configs = sum([1 for x in [user_config_file, config_content, config_dict] if x is not None])
+
+        if configs > 1:
+            raise ValueError("Attempting to construct SiteConfig with multiple config objects")
+
+        if configs == 0:
+            raise ValueError("Attempting to construct SiteConfig with no config objects")
+
+        c_ptr = None
         if user_config_file is not None:
-            if not isfile(user_config_file):
+            if not os.path.isfile(user_config_file):
                 raise IOError('No such configuration file "%s".' % user_config_file)
-
             c_ptr = self._alloc_load_user_config(user_config_file)
-            if c_ptr:
-                super(SiteConfig, self).__init__(c_ptr)
-            else:
-                raise ValueError('Failed to construct SiteConfig instance from config file %s.' % user_config_file)
 
-        else:
+        elif config_content is not None:
             c_ptr = self._alloc(config_content)
 
-            if c_ptr is None:
-                raise ValueError('Failed to construct SiteConfig instance.')
-
-            super(SiteConfig, self).__init__(c_ptr)
+        elif config_dict is not None:
+            license_root_path = config_dict.get(ConfigKeys.LICENSE_PATH)
+            license_root_path_site = os.path.realpath(license_root_path)
+            __license_root_path = os.path.join(
+                license_root_path_site,
+                os.getenv("USER"),
+                str(os.getpid())
+            )
+
+            # Create joblist
+            ext_job_list = ExtJoblist()
+            for job in config_dict.get(ConfigKeys.INSTALL_JOB, []):
+                if not os.path.isfile(job['PATH']):
+                    print("Unable to locate job file {}".format(job['PATH']))
+                    continue
+                new_job = ExtJob(config_file=job['PATH'], private=False, name=job['NAME'], license_root_path=__license_root_path)
+                new_job.convertToCReference(None)
+                ext_job_list.add_job(job['NAME'], new_job)
+
+            for job_path in config_dict.get(ConfigKeys.INSTALL_JOB_DIRECTORY, []):
+                if not os.path.isdir(job_path):
+                    print("Unable to locate job directory {}".format(job['PATH']))
+                    continue
+                files = os.listdir(job_path)
+                for file_name in files:
+                    full_path = os.path.join(job_path, file_name)
+                    try:
+                        new_job = ExtJob(config_file=full_path, private=False, license_root_path=__license_root_path)
+                        new_job.convertToCReference(None)
+                        ext_job_list.add_job(new_job.name(), new_job)
+                    except:
+                        continue
```

Comment:
is this exception ok without any logging / printing ? or should we have something which notice us about the job creation is failing..?

Reply:
You are right, should be something to warn the user..